lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1221 @@
+      new BCNLCommand(BCNLMessage.ASCII_CONTROL.EOT).send(this.hidDevice)
+        .then(response => response.checkAsciiControl(BCNLMessage.ASCII_CONTROL.ENQ))
+        .then(() => resolve())
+        .catch(err => reject(err));
+    });
+  }
+
+  togglePassthroughMode(mode) {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      new BCNLCommand('W|').send(this.hidDevice)
+        .then(response => response.checkAsciiControl(BCNLMessage.ASCII_CONTROL.ACK))
+        .then(() => new BCNLCommand('Q|').send(this.hidDevice))
+        .then(response => response.checkAsciiControl(BCNLMessage.ASCII_CONTROL.ACK))
+        .then(() => new BCNLCommand(`${mode}|`).send(this.hidDevice))
+        .then(response => response.checkAsciiControl(BCNLMessage.ASCII_CONTROL.ACK))
+        .then(() => resolve())
+        .catch(err => reject(err));
+    });
+  }
+
+  openConnection() {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      new OpenConnectionRequest(this.pumpSession).send(this.hidDevice)
+        // eslint-disable-next-line no-unused-vars
+        .then((response) => {
+          // TODO - do we care about the response?
+          resolve();
+        });
+    });
+  }
+
+  readPumpInfo() {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      new ReadInfoRequest(this.pumpSession).send(this.hidDevice)
+        .then((response) => {
+          this.pumpSession.linkMAC = response.linkMAC;
+          this.pumpSession.pumpMAC = response.pumpMAC;
+          resolve();
+        });
+    });
+  }
+
+  getLinkKey() {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      new RequestLinkKeyRequest(this.pumpSession).send(this.hidDevice)
+        .then((response) => {
+          this.pumpSession.key = response.linkKey(this.pumpSession.modelAndSerial);
+          resolve();
+        });
+    });
+  }
+
+  negotiateRadioChannel() {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      // TODO - reorder the channel list if we know the last used radio channel
+      async.eachSeries(
+        MM600SeriesDriver.CHANNELS,
+        (channel, callback) => {
+          this.pumpSession.radioChannel = channel;
+          new JoinNetworkRequest(this.pumpSession).send(this.hidDevice)
+            .then((response) => {
+              if (response.joinedNetwork) {
+                callback(true);
+              } else {
+                callback(null);
+              }
+            })
+            .catch((err) => {
+              callback(err);
+            });
+        },
+        (result) => {
+          if (result instanceof Error) {
+            throw result;
+          } else {
+            if (result === null) {
+              this.pumpSession.radioChannel = 0;
+              reject(new Error('Cannot connect to pump. Are you nearby?'));
```

Comment:
You're right. The wording here is terrible. Will fix :) Errors are surfacing in the UI, but only in the "Error details" section. I'll look to make user-actionable messages like this more obvious.